src/main/java/org/spongepowered/common/conversation/SpongeConversationManager.java

```diff
@@ -0,0 +1,192 @@
+                final Optional<Question> optionalQuestion = conversation.getCurrentQuestion();
+                if (!optionalQuestion.isPresent()) {
+                    conversant.sendThroughMessage(Text.of("Your current conversation has no current question. Feel free to attempt to leave it."));
+                } else {
+                    final CommandContext context = new CommandContext();
+                    final Question question = optionalQuestion.get();
+                    try {
+                        question.getArguments()
+                            .parse(conversant, new CommandArgs(message, InputTokenizer.quotedStrings(false).tokenize(message, false)), context);
+                        final QuestionResult questionResult = question.getHandler().handle(conversation, conversant, question, context);
+                        switch (questionResult.getType()) {
+                            case NEXT:
+                                Optional<Question> nextQuestion = questionResult.getNextQuestion();
+                                if (!nextQuestion.isPresent()) {
+                                    this.logger.error(conversant.getName() + "'s next question was missing. Their conversation is ending.");
+                                    conversant.sendThroughMessage(Text.of("Your next question is missing, ending conversation."));
+                                    conversation.end(ConversationEndTypes.ERROR, Cause.of(NamedCause.source(conversation.getCreator())));
+                                    return true;
+                                }
+                                conversation.setQuestion(nextQuestion.get());
+                                break;
+                            case REPEAT:
+                                conversation.setQuestion(question);
+                                break;
+                            case END:
+                                conversation.end(ConversationEndTypes.FINISHED, Cause.of(NamedCause.source(conversation.getCreator())));
+                                break;
+                            default:
+                                // Should not occur
+                                SpongeImpl.getLogger().error("For some reason the question result type did not match any.");
+                                conversation.end(ConversationEndTypes.ERROR, Cause.of(NamedCause.source(SpongeImpl.getPlugin())));
+                        }
+                    } catch (ArgumentParseException e) {
+                        conversant.sendThroughMessage(Text.of("Failed to parse your specified arguments. Restarting question."));
+                        conversation.setQuestion(question);
+                        SpongeImpl.getLogger().error("Failed to parse the arguments passed into " + conversant.getName() +
+                            "'s conversation." + " Sending them the same question again.");
+                        e.printStackTrace();
```

Comment:
You should give the exception to the logger message above and use placeholders instead of building the message manually. Basically like:

```java
SpongeImpl.getLogger().error("Failed to parse the arguments passed into {}'s conversation. Sending them the same question again.", conversant.getName(), e);
```